Activities/Module13 Activity.md

```diff
@@ -146,3 +146,219 @@ In case your server is is not responding to requests after a long period of inac
+### Step 1: Enroll in GitHub Student Developer Pack and Heroku for Students
+
+1. **Enroll in GitHub Student Developer Pack**
+   * Visit GitHub Education and select **"Get student benefits"**. Follow the instructions to apply. Approval can take up to 2 business days.
+
+2. **Set Up Heroku for Students**
+   * Go to Heroku.com and create an account (use the same email as your GitHub account).
+   * Enroll in the Heroku for Students offer, which requires the GitHub Student Developer Pack.
+      * **Note**: You'll need to add a credit card for the platform credits that Heroku provides for student use.
+   * After approval, you should see these credits in the billing section of your Heroku account within a few hours. **Wait to create an app until credits are available.**
+
+### Step 2: Create a New Application on Heroku
+
+1. **Navigate to the Heroku Dashboard**
+   * After logging into Heroku, go to your Dashboard.
+
+2. **Create a New App**
+   * Select **New -> Create New App**.
+   * Choose a descriptive app name (this will form part of your app's URL).
+   * Click **Create App** to proceed.
+
+   ![Create New App in Heroku](Module13Assets/CreateNewApp.png)
+
+### Step 3: Deploy the Server from a New Repository
+
+1. **Create a New Private Repository**
+   * You must create a new Private Repository on Github which will contain only your server code.
+   * Create a new Private Repository on Github and copy paste your server folder.
+   * Copy the .gitignore from the root of the original repository to the server repository.
+
+2. **Update Yaml file**
+   * Once the client and server folders are separated into different repositories, update the .github\workflows\main.yml in each as the following:
+
+   **Client - GitHub Actions CI Workflow Configuration**
+
+      ```yaml
+      name: FakeStackOverflow CI
+      on: # Controls when the action will run.
+      # Triggers the workflow on push or pull request events but only for the master branch. If you want to trigger the action on other branches, add here
+      push:
+         branches: [main]
+      pull_request:
+         branches: [main]
+
+      # Allows you to run this workflow manually from the Actions tab
+      workflow_dispatch:
+
+      # A workflow run is made up of one or more jobs that can run sequentially or in parallel
+      jobs:
+      build-and-test: #
+         # The type of runner that the job will run on
+         runs-on: ubuntu-20.04
+
+         # Steps represent a sequence of tasks that will be executed as part of the job
+         steps:
+            - name: Git Checkout
+            uses: actions/checkout@v4
+
+            - name: Set up Node.js 20.x
+            uses: actions/setup-node@v4
+            with:
+               node-version: "20.x"
+
+            - name: Build and lint frontend
+            if: ${{ always() }}
+            env:
+               REACT_APP_SERVER_URL: http://localhost:8000
+            run: cd client; npm ci && npm run build && npm run lint
```

Comment:
Once the repos are separated, will `cd client` still apply?

Reply:
Yes, because on the original GitHub classroom repo we are not changing the file structure.